core/src/main/java/eu/fasten/core/maven/MavenResolver.java

```diff
@@ -222,41 +222,43 @@ public void run() {
      * @param timestamp Timestamp for filtering dependency versions (-1 in order not to filter)
      * @param dbContext Database connection context (needed only for filtering by timestamp)
      * @return A dependency set (including all transitive dependencies) of given Maven coordinate
-     * @throws NoSuchElementException if pom file was not retrieved
      */
     public Set<Revision> resolveFullDependencySetOnline(String group, String artifact, String version,
-                                                          long timestamp, DSLContext dbContext) throws FileNotFoundException {
+                                                          long timestamp, DSLContext dbContext) {
         Set<Revision> dependencySet = new HashSet<>();
-
-        logger.debug("Downloading artifact's POM file");
-
-        var pomOpt = MavenUtilities.downloadPom(group, artifact, version);
-        var pom = pomOpt.orElseGet(() -> {
-            throw new RuntimeException("Could not download pom file");
-        });
-        if (!pom.exists()) {
-            throw new FileNotFoundException("Could not find file: " + pom.getAbsolutePath());
+        File tempDir;
+        try {
+            tempDir = Files.createTempDirectory("fasten-pom").toFile();
+        } catch (IOException e) {
+            logger.error("Unable to create a temp directory", e);
+            return null;
         }
-
-        File outputFile = null;
         try {
             logger.debug("Running 'mvn dependency:list'");
-
-            outputFile = Files.createTempFile("fasten", ".txt").toFile();
-
-            String[] cmd = new String[]{
-                    "mvn",
-                    "dependency:list",
-                    "-f", "\"" + pom.getAbsolutePath() + "\"",
-                    "-DoutputFile=" + outputFile.getAbsolutePath()
-            };
-            var process = new ProcessBuilder(cmd).start();
-            final var timeoutSeconds = 30;
-            var completed = process.waitFor(timeoutSeconds, TimeUnit.SECONDS);
+            var cmd = new String[]{
```

Comment:
Maybe I'm missing something, but why is `junit` hardcoded in this code?

Reply:
Oops, my bad :)
Must have left it there accidentally after testing. Will fix!